mmcv/transform/loading.py

```diff
@@ -0,0 +1,241 @@
+    provides.
+
+    The annotation format as follow:
+    {
+        'instances':
+        [
+            {
+            # List of 4 numbers representing the bounding box of the
+            # instance, in (x1, y1, x2, y2) order.
+            'bbox': [x1, y1, x2, y2],
+
+            # Label of image classification.
+            'bbox_label': 1,
+
+            # Used in key point detection.
+            # In the format of [x1, y1, v1,…, xn, yn, vn]. v[i] means the
+            # visibility of this keypoint. n must be equal to the number
+            # of keypoint categories.
+            'keypoints': [x1, y1, v1, ..., xn, yn, vn]
+            }
+        ]
+        # Filename of semantic or panoptic segmentation ground truth file.
+        'seg_map': 'a/b/c'
+    }
+    After load and process:
+    {
+        'gt_bboxes': np.array([[x1, y1, x2, y2]])
+        'gt_bboxes_labels': np.array([1])
+        'gt_semantic_seg': np.array()
+        'gt_keypoints': np.array([[[x1, y1, v1], [xn, yn, vn]]])
+    }
+
+    Args:
+        with_bbox (bool): Whether to parse and load the bbox annotation.
+             Default: True.
+        with_label (bool): Whether to parse and load the label annotation.
+            Default: True.
+        with_seg (bool): Whether to parse and load the semantic segmentation
+            annotation. Default: False.
+        with_kps (bool): Whether to parse and load the keypoints annotation.
+             Default: False.
+        imdecode_backend (str): The image decoding backend type. The backend
+            argument for :func:`mmcv.imfrombytes`.
+            See :fun:`mmcv.imfrombytes` for details.
+            Defaults to 'cv2'.
+        file_client_args (dict): Arguments to instantiate a FileClient.
+            See :class:`mmcv.fileio.FileClient` for details.
+            Defaults to ``dict(backend='disk')``.
+    """
+
+    def __init__(
+        self,
+        with_bbox: bool = True,
+        with_label: bool = True,
+        with_seg: bool = False,
+        with_kps: bool = False,
+        imdecode_backend: str = 'cv2',
+        file_client_args: dict = dict(backend='disk')
+    ) -> None:
+        super().__init__()
+        self.with_bbox = with_bbox
+        self.with_label = with_label
+        self.with_seg = with_seg
+        self.with_kps = with_kps
+        self.imdecode_backend = imdecode_backend
+        self.file_client_args = file_client_args.copy()
+        self.file_client = None
+
+    def _load_bboxes(self, results: dict) -> dict:
+        """Private function to load bounding box annotations.
+
+        Args:
+            results (dict): Result dict from :obj:`mmcv.BaseDataset`.
+        Returns:
+            dict: The dict contains loaded bounding box annotations.
+        """
+        gt_bboxes = []
+        for instance in results['instances']:
+            gt_bboxes.append(instance['bbox'])
+        results['gt_bboxes'] = np.array(gt_bboxes)
+        return results
```

Comment:
I thinks whether or not we need to return `results`, since we change `results` inplace.